include/llzk/Dialect/Struct/IR/Ops.td

```diff
@@ -162,29 +193,41 @@ def LLZK_FieldDefOp
     ```llzk
     struct.field @f1 : !felt.type
     struct.field @f2 : !felt.type {llzk.pub}
+    struct.field @col1 : !felt.type {column}
+    struct.field @sig1 : !felt.type {signal}
+    struct.field @colsig1 : !felt.type {column, signal}
+    struct.field @pubcol : !felt.type {llzk.pub, column}
     ```
+
+    - Fields marked with the `{llzk.pub}` attribute are considered public
+      and represent outputs of their defining struct/component.
+    - Fields marked with the `{column}` attribute can be read/written with
+      table offsets using the `readf` and `writef` operations.
+    - Fields marked with the `{signal}` attribute are constraint variables that
+      are stored in the witness; non-signal values are intermediate expressions.
   }];
 
   let arguments = (ins SymbolNameAttr:$sym_name, TypeAttrOf<AnyLLZKType>:$type,
-      UnitAttr:$column);
+      UnitAttr:$column, UnitAttr:$signal);
 
   // Define builders manually to avoid the default ones that have extra
   // TypeRange parameters that must always be empty.
   let skipDefaultBuilders = 1;
   let builders =
       [OpBuilder<(ins "::mlir::StringAttr":$sym_name, "::mlir::TypeAttr":$type,
-           CArg<"bool", "false">:$isColumn)>,
+           CArg<"bool", "false">:$isColumn, CArg<"bool", "false">:$isSignal)>,
```

Comment:
In all of the builders, can you order signal before column. Since they both have default values and it seems a more common scenario to set `isSignal=true` than `isColumn=true`.